provider/reprovider.go

```diff
@@ -383,7 +383,7 @@ func (s *reprovider) run() {
 
 			s.throughputDurationSum += dur
 			s.throughputProvideCurrentCount += uint(len(keys))
-			if s.throughputCallback != nil && s.throughputProvideCurrentCount >= s.throughputMinimumProvides {
+			if s.throughputCallback != nil && s.reprovideInterval > 0 && s.throughputProvideCurrentCount >= s.throughputMinimumProvides {
```

Comment:
@guillaumemichel hm.. but this code would execute only if  `performedReprovide` is `true`, so we are not fixing underlying problem, just hiding error message, or did I misread whats happening? 

any idea why `performedReprovide` would be `true` if reprovide interval is `0`?

Reply:
Found the cause and addressed it in https://github.com/ipfs/boxo/pull/871/commits/01710006ac0c1991039532d5309156ec4df1eb38